docs/api.md

```diff
@@ -61,15 +61,15 @@ Properties not defined in the schema will not be included in serialized JSON, un
 If you use JTD with typescript, the type for the schema can be derived from the data type, and generated serializer would only accept correct data type in this case:
 
 ```typescript
-import Ajv, {JTDSchemaType} from "ajv/dist/jtd"
+import Ajv from "ajv/dist/jtd"
 const ajv = new Ajv()
 
-interface MyData = {
+interface MyData {
   foo: number
   bar?: string
 }
 
-const mySchema: JTDSchemaType<MyData> = {
```

Comment:
... but without explicitly defining schema type, the serializer would not check data type, as the comment below says - why is it removed here?

Reply:
When I use typescript code below, tsc couldn't compile it.
```typescript
import Ajv, {JTDSchemaType} from "ajv/dist/jtd"
const ajv = new Ajv()

interface MyData {
  foo: number
  bar?: string
}

const mySchema: JTDSchemaType<MyData> = {
  properties: {
    foo: {type: "int32"} // any JTD number type would be accepted here
  },
  optionalProperties: {
    bar: {type: "string"}
  }
}

const serializeMyData = ajv.compileSerializer(mySchema)
```

```bash
$ npx ts-node src/main.ts
⨯ Unable to compile TypeScript:
src/main.ts:11:11 - error TS2322: Type 'string' is not assignable to type 'never'.

11     foo: {type: "int32"} // any JTD number type would be accepted here
             ~~~~
src/main.ts:14:11 - error TS2322: Type 'string' is not assignable to type 'never'.

14     bar: {type: "string"}
             ~~~~
```


But when I use code below, tsc could compile it. 
Plus I thought the result may be correct.
I'm not so confident since I am beginner here.

```typescript
import Ajv, {JTDSchemaType} from "ajv/dist/jtd"
const ajv = new Ajv()

interface MyData {
  foo: number
  bar?: string
}

// const mySchema: JTDSchemaType<MyData> = {
const mySchema = {
  properties: {
    foo: {type: "int32"} // any JTD number type would be accepted here
  },
  optionalProperties: {
    bar: {type: "string"}
  }
}

const serializeMyData = ajv.compileSerializer(mySchema)
console.log(serializeMyData({"foo": "123", "baz": 123})); // => {"foo":123}
console.log(serializeMyData({"foo": "123"})); // => {"foo":123}
console.log(serializeMyData({})); // => {"foo":undefined}
```

Reply:
There can be two reasons - typescript prior to 4.2.3 or not set strictNullChecks

Reply:
From the error message the latter seems more likely

Reply:
I tried unset strictNullChesk. But it didn't work.

tsconfig.json
```json
{
  "compilerOptions": {
    "strictNullChecks": false
  }
}
```

```bash
$ npx ts-node --project /path/to/tsconfig.json src/main.ts
⨯ Unable to compile TypeScript:
src/main.ts:12:11 - error TS2322: Type 'string' is not assignable to type 'never'.

12     foo: {type: "int32"} // any JTD number type would be accepted here
             ~~~~
src/main.ts:15:11 - error TS2322: Type 'string' is not assignable to type 'never'.

15     bar: {type: "string"}
             ~~~~
```

I tried to use older version of typescript. It didn't work either.
Since I can't understand why, should I revert the type change of mySchema ?

Reply:
strictNullChecks should be true - I will check the example a bit later

Reply:
Typescript should be at least 4.2.3, not older

Reply:
You can also use it without schema type, and explicitly pass type parameter to compileSerializer

Reply:
Thank you for your help. It worked.

```json
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

```typescript
import Ajv, {JTDSchemaType} from "ajv/dist/jtd"
const ajv = new Ajv()

interface MyData {
  foo: number
  bar?: string
}

const mySchema: JTDSchemaType<MyData> = {
// const mySchema = {
  properties: {
    foo: {type: "int32"} // any JTD number type would be accepted here
  },
  optionalProperties: {
    bar: {type: "string"}
  }
}

const serializeMyData = ajv.compileSerializer(mySchema)
console.log(serializeMyData({foo: 123, bar: "str"})); // => {"foo":123,"bar":"str"}
console.log(serializeMyData({foo: 123})); // => {"foo":123}
```

I' fix the code change.